.circleci/config.yml

```diff
@@ -1,32 +1,47 @@
 version: 2
 
-jobs:
-  checkout_code:
+# This file uses YAML anchors to deduplicate steps
+# see https://circleci.com/blog/circleci-hacks-reuse-yaml-in-your-circleci-config-with-yaml/
+# and https://learnxinyminutes.com/docs/yaml/
+
+templates:
+  job_template: &job_template
     docker:
-      - image: &builder_image jfullaondo/datadog-agent-builder-deb:0.0.4
+      - image: jfullaondo/datadog-agent-builder-deb:0.0.4
+        environment:
+          USE_SYSTEM_LIBS: "1"
     working_directory: /go/src/github.com/DataDog/datadog-agent
+  step_templates:
+    - run: &save_sha
+        name: save SHA to a file
+        command: echo $CIRCLE_SHA1 > .circle-sha
+    - restore_cache: &restore_source
+        keys:
+          - v1-repo-{{ checksum ".circle-sha" }}
+    - restore_cache: &restore_vendor
+        keys:
+          - v1-govendordeps-{{ checksum "Gopkg.toml" }}
+    - restore_cache: &restore_gobindeps
+        keys:
+          - v1-gobindeps
+
+jobs:
+  checkout_code:
+    <<: *job_template
     steps:
       - checkout
-      - run:
-          name: save SHA to a file
-          command: echo $CIRCLE_SHA1 > .circle-sha
+      - run: *save_sha
       - save_cache:
           key: v1-repo-{{ checksum ".circle-sha" }}
           paths:
             - /go/src/github.com/DataDog/datadog-agent
 
   dependencies:
-    docker:
-      - image: *builder_image
-    working_directory: /go/src/github.com/DataDog/datadog-agent
+    <<: *job_template
     steps:
-      - run:
-          name: save SHA to a file
-          command: echo $CIRCLE_SHA1 > .circle-sha
-      - restore_cache:
-          keys:
```

Comment:
BTW, if several keys are provided, cci does not restore both, but restores the first key lookup that succeeds. That means we were not restoring the vendor dir before running inv deps. Granted, that's not saving us time for now as most of the time is scanning the sources for deps: https://circleci.com/gh/DataDog/datadog-agent/5186
Next improvement would be to correctly handle the lockfile and use only that to install deps